invenio_app_rdm/records_ui/views/deposits.py

```diff
@@ -479,6 +479,7 @@ def deposit_create(community=None):
         searchbar_config=dict(searchUrl=get_search_url()),
         record=new_record(),
         community=community,
+        community_ui=community,
```

Comment:
this is a little bit confusing comparing with the line above, could you try to get community_ui representation and assign it here instead of the community object? I think it was done in other community related views (for example check requests_ui)

Reply:
the UI serialization already happens in the decorator: https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/views/decorators.py#L349-L367

Reply:
argh.... can we do the opposite then and in community assign the actual community, not UI serialized one? @jrcastro2 already made the attempt to normalise these around, I think we missed this one

Reply:
if i'm not mistaken, the UI-serialized stuff is just a dictionary?
as such, i'll edit the decorator to give `community` as API object, and `community_ui` as UI-serialized?

Reply:
if i understand correctly, this is what i've added now.
or do you mean we should only use the `@pass_community` decorator and toss out the `@pass_draft_community`?

Reply:
I mean when we pass variables to jinja context, we should avoid the naming confusion
so when we pass 
```
community_ui=community
community=community
```
from the naming I would expect to have two different community objects in the template context, while I only get the UI one.

so either we should remove `community=community` entirely (potentially with some consequences) or do
```
community_ui=community_ui # ui serialized object
community=community # community service result item object
```
pinging @jrcastro2 for reference on the same changes on record and record_ui

Reply:
to be clear, i *have* updated this PR to have `community=<result-item>`, and `community_ui=<ui-serialized-item>`
if i happened to miss one, could you please point it out in the lines?

Reply:
Recently we agreed to not mix the object with the UI dict, in the past we were calling it record and it could be the record_ui, which is confusing because sometimes you need the actual object and not the dict. So we fixed it in for the record in this PR: https://github.com/inveniosoftware/invenio-app-rdm/commit/3b8cbc4a06db2a32dcd5e30309cc251a8613cc5b#diff-2fb602eed821cbd7d07f7c61560b3e8ed50e165259da085c113fff47274b8496R236-R237

Now with the latest changes it looks good to me 👍

Reply:
> if i understand correctly, this is what i've added now. or do you mean we should only use the `@pass_community` decorator and toss out the `@pass_draft_community`?

No, it's more about what type of object goes in `community` and `community_ui`,  the `@pass_draft_community` is needed since it gets the community from the querystring.